wgsl_to_wgpu/src/lib.rs

```diff
@@ -64,6 +101,76 @@ pub enum CreateModuleError {
     /// Each binding resource must be associated with exactly one binding index.
     #[error("duplicate binding found with index `{binding}`")]
     DuplicateBinding { binding: u32 },
+
+    /// The shader source could not be parsed.
+    #[error("failed to parse: {error}")]
+    ParseError {
+        error: naga::front::wgsl::ParseError,
+    },
+
+    /// The shader source could not be validated.
+    #[error("failed to validate: {error}")]
+    ValidationError {
+        error: WithSpan<naga::valid::ValidationError>,
+    },
+}
+
+impl CreateModuleError {
+    /// Writes a diagnostic error to stderr.
+    pub fn emit_to_stderr<P: AsRef<Path>>(&self, wgsl_source: &str, path: Option<P>) {
```

Comment:
The compiler is unable to infer the generic type when passing in `None`. It might be better to just add a separate `emit_to_stderr_with_path` method like naga.